pkg/controller/utils/utils.go

```diff
@@ -299,3 +300,22 @@ func GetExpectedTyphaScale(nodes int) int {
 	}
 	return typhas
 }
+
+// GetElasticLicenseType returns the Elasticsearch license type from elastic-licensing configmap that ECK operator keeps updated.
+// If elastic-licensing configmap doesn't exists, it must be a fresh install, return "basic" license.
+func GetElasticLicenseType(ctx context.Context, cli client.Client) (render.ElasticLicenseType, error) {
+	cm := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{Name: render.ECKLicenseConfigMapName, Namespace: render.ECKOperatorNamespace}, cm)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
Does the ECK operator create a basic license config map if there's no applied config map? If it does, then you should return "Unknown" here, if it doesn't, then ignore this comment.

Reply:
For fresh install of caliEnt, elasticsearch license will be basic by default, plus during the fresh install we won't have any existing `elastic-licensing` configmap until ECK operator is running, so we return "basic" if config map is not found.
Once ECK operator is running, it always creates this config map irrespective of license type. Only possible scenario where config map is not found is during fresh install.

Reply:
> Once ECK operator is running, it always creates this config map irrespective of license type

With this in mind I think we should not proceed on anything that requires having knowledge of the license, let's wait until the ECK operator creates the license.

When we have the option to we should avoid assuming assuming configuration for third party software, in this case we just need to wait for ECK to tell us what configuration it will apply. If it changes in the future we won't have a scenario where we temporarily use a basic license then immediately switch to a different one.